src/player.h

```diff
@@ -594,6 +594,7 @@ class Player final : public Creature, public Cylinder
 		bool isImmune(ConditionType_t type) const final;
 		bool hasShield() const;
 		bool isAttackable() const final;
+		bool lastHitIsPlayer(Creature* lastHitCreature) const;
```

Comment:
Alright. I have never heard of the expression 'free function' before, are you simply referring to a non-member function that you only declare inside the cpp file? If so, what is the preferred method to do so? Declare the function before the member function where it's being utilized, or at the bottom (and provide a function prototype for the function at the top).